commit b4b171c4918d39c38d03bafe98bc52684e7927e1
Author: Nils Philippsen <nils@redhat.com>
Date:   Mon Jun 22 18:25:01 2009 +0200

    patch: hal
    
    Squashed commit of the following:
    
    commit 5a4809b0c721ac14b8e56064ee3df573e75e0b67
    Author: Nils Philippsen <nils@redhat.com>
    Date:   Mon Jun 22 18:23:27 2009 +0200
    
        separate HAL information and policy
    
    commit a50c619ddbb869612e99a9457b1f39492f34b141
    Author: Nils Philippsen <nils@redhat.com>
    Date:   Tue Jun 16 15:21:59 2009 +0200
    
        adapt HAL fdi files for Fedora

diff --git a/tools/Makefile.am b/tools/Makefile.am
index bd6dd9d..bb26fcb 100644
--- a/tools/Makefile.am
+++ b/tools/Makefile.am
@@ -79,7 +79,7 @@ udev/libsane.rules: $(wildcard ${top_srcdir}/doc/descriptions/*.desc) $(wildcard
 	   -d 0 > $@
 
 hal/libsane.fdi: $(wildcard ${top_srcdir}/doc/descriptions/*.desc) $(wildcard ${top_srcdir}/doc/descriptions-external/*.desc) sane-desc
-	@./sane-desc -m hal -s ${top_srcdir}/doc/descriptions:${top_srcdir}/doc/descriptions-external \
+	@./sane-desc -m hal-new -s ${top_srcdir}/doc/descriptions:${top_srcdir}/doc/descriptions-external \
 	   -d 0 > $@
 
 sane-backends.pc:	$(top_builddir)/config.status
diff --git a/tools/hal/libsane-policy.fdi b/tools/hal/libsane-policy.fdi
new file mode 100644
index 0000000..2ac8211
--- /dev/null
+++ b/tools/hal/libsane-policy.fdi
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<deviceinfo version="0.2">
+  <device>
+    <match key="info.subsystem" string="usb">
+      <!-- The following rule will disable autosuspend for USB scanner devices -->
+      <match key="info.capabilities" contains="scanner">
+        <append key="info.callouts.add" type="strlist">hal-usb-scanner-add</append>
+      </match>
+    </match>
+  </device>
+</deviceinfo>
diff --git a/tools/sane-desc.c b/tools/sane-desc.c
index 600ab03..c317700 100644
--- a/tools/sane-desc.c
+++ b/tools/sane-desc.c
@@ -3667,7 +3667,6 @@ print_hal (int new)
       printf (" -->\n");
       printf ("        <match key=\"usb.product_id\" int=\"%s\">\n", usbid->usb_product_id);
       printf ("          <append key=\"info.capabilities\" type=\"strlist\">scanner</append>\n");
-      printf ("          <merge key=\"scanner.access_method\" type=\"string\">proprietary</merge>\n");
       printf ("        </match>\n");
 
       usbid = usbid->next;
